libtenzir/builtins/formats/json.cpp

```diff
@@ -158,11 +158,19 @@ class doc_parser {
 
 private:
   auto report_parse_err(auto& v, std::string description) -> void {
+    if (v.current_location().error()) {
+      ctrl_.warn(
+        caf::make_error(ec::parse_error,
+                        fmt::format("failed to parse {} in line '{}'",
+                                    std::move(description), parsed_document_)));
+      return;
+    }
+    const auto faulty_json_view = std::string_view{
+      v.current_location().value_unsafe(), parsed_document_.end()};
```

Comment:
We _could_ add a cut off and print `...` after a certain limit, but I think that we should just not emit such diagnostics. So I'd rather see some logic here to emit a good diagnostic. For now it perhaps suffices to add the `...` here after some reasonable number of characters (e.g., max 30), but I feel like seeing just the remaining string might not be too helpful in general.

Reply:
@patszt can you make one of the following two changes:
1. Drop the unparsed section of the document.
2. Truncate (if the length is over 30 bytes) with a trailing `…` and JSON escape (always).

Reply:
I did truncate it to 50. I think the version where we print line/column is fine.
The other part is also not that informative with 50 bytes of information.
What would be nice is:
- Some bytes before current_location(). It would be nice to display a one field before, but it can always be a long nested field.
- A whole JSON body with the faulty part being marked with some underline  /\/\/\. Due to it being big i'd like it to be hidden by default and have an option to show it on the UI somehow